src/components/date-picker-month-header/date-picker-month-header.tsx

```diff
@@ -259,27 +264,50 @@ export class DatePickerMonthHeader {
     this.calciteDatePickerSelect.emit(date);
   };
 
-  /**
-   * Parse localized year string from input,
-   * set to active if in range
-   */
-  private setYear(localizedYear: string, increment = 0) {
-    const { min, max, activeDate, localeData, yearInput } = this;
+  private getInRangeDate({
+    localizedYear,
+    increment = 0
+  }: {
+    localizedYear: string;
+    increment?: number;
+  }): Date {
+    const { min, max, activeDate, localeData } = this;
     const parsedYear = parseNumber(localizedYear, localeData);
     const length = parsedYear.toString().length;
     const year = isNaN(parsedYear) ? false : parsedYear + increment;
     const inRange =
       year && (!min || min.getFullYear() <= year) && (!max || max.getFullYear() >= year);
-    // if you've supplied a year and it's in range, update active date
+    // if you've supplied a year and it's in range
     if (year && inRange && length === localizedYear.length) {
       const nextDate = new Date(activeDate);
       nextDate.setFullYear(year as number);
-      const inRangeDate = dateFromRange(nextDate, min, max);
+      return dateFromRange(nextDate, min, max);
+    }
+  }
+
+  /**
+   * Parse localized year string from input,
+   * set to active if in range
+   */
+  private setYear({
+    localizedYear,
+    reset = true,
+    increment = 0
+  }: {
+    localizedYear: string;
+    reset?: boolean;
+    increment?: number;
+  }) {
+    const { yearInput, activeDate, localeData } = this;
+    const inRangeDate = this.getInRangeDate({ localizedYear, increment });
+
+    // if you've supplied a year and it's in range, update active date
+    if (inRangeDate) {
       this.calciteDatePickerSelect.emit(inRangeDate);
-      yearInput.value = localizeNumber(inRangeDate.getFullYear(), localeData);
-    } else {
-      // leave the current active date and clean up garbage input
-      yearInput.value = localizeNumber(activeDate.getFullYear(), localeData);
+    }
+
+    if (reset) {
```

Comment:
Nitpick: in terms of naming, is this behavior more of a `commit` than a `reset`?